### PR TITLE
Task/bws156 output os for table

### PR DESCRIPTION
**Pull request by tbezrukava**
Снимок экрана 2022-03-21 в 13 21 16

### CODE REVIEW

**Review by what1s1ove — DISMISSED** [review]

File: backend/src/data/repositories/instance/instance.repository.ts [hunk 1]
```diff
@@ -21,8 +21,11 @@ class Instance {
       state?: InstanceState;
       hostname?: string;
     },
-  ): Promise<InstanceEntity> {
-    return this.#InstanceModel.query().patchAndFetchById(id, data);
+  ): Promise<InstanceM> {
```
Comment (Yaroslaww-1) [inline backend/src/data/repositories/instance/instance.repository.ts]:
```suggestion
  ): Promise<InstanceEntity> {
```

File: backend/src/data/repositories/instance/instance.repository.ts [hunk 2]
```diff
@@ -21,8 +21,11 @@ class Instance {
       state?: InstanceState;
       hostname?: string;
     },
```
Comment (Yaroslaww-1) [inline backend/src/data/repositories/instance/instance.repository.ts]: Ideally method should look like that:
```
public async updateById(instance: InstanceEntity): Promise<void>
```

File: backend/src/data/repositories/instance/instance.repository.ts [hunk 3]
```diff
@@ -58,17 +61,16 @@ class Instance {
   }: {
     filter: SCInstanceGetByTenantRequestParamsDto;
     tenantId: string;
-  }): Promise<InstanceEntity[]> {
+  }): Promise<InstanceM[]> {
```
Comment (Yaroslaww-1) [inline backend/src/data/repositories/instance/instance.repository.ts]:
```suggestion
  }): Promise<InstanceEntity[]> {
```
You can add operationSystem property yo your entity

**Review by Yaroslaww-1 — CHANGES_REQUESTED** [review]

**Review by ania-leichenko — DISMISSED** [review]

File: backend/src/services/operation-system/operation-system.service.ts [hunk 4]
```diff
@@ -1,6 +1,8 @@
 import { operationSystem as OperationSystemRep } from '~/data/repositories/repositories';
 import { OperationSystem as OperationSystemEntity } from '~/services/operation-system/operation-system.entity';
 import { SCOperationSystemGetAllResponseDto } from '~/common/types/types';
+import { SCError } from '~/exceptions/sc-error/sc-error';
```
Comment (what1s1ove) [inline backend/src/services/operation-system/operation-system.service.ts]:
```suggestion
import { SCError } from '~/exceptions/exceptions';
```

**Review by what1s1ove — CHANGES_REQUESTED** [review]

**Review by what1s1ove — DISMISSED** [review]

File: backend/src/services/instance/instance.entity.ts [hunk 5]
```diff
@@ -13,6 +14,7 @@ class Instance {
   public awsInstanceId: string;
   public tenantId: string;
   public state: InstanceState;
+  public operationSystem: SCInstanceRelatedItem;
```
Comment (Yaroslaww-1) [inline backend/src/services/instance/instance.entity.ts]: Dto is used **only** for sharing API responses types with frontend. You should never use dto for any other types. If you need backend specific enums/types you must create them on backend only. You can either use src/common/enums folder or src/services/instance

**Review by Yaroslaww-1 — CHANGES_REQUESTED** [review]

File: frontend/src/components/sc/components/instances-table/helpers/get-columns.helper.ts [hunk 6]
```diff
@@ -24,6 +24,10 @@ const getColumns = (): Column[] => {
       minWidth: 100,
       width: 150,
     },
+    {
+      Header: InstancesTableHeader.OS,
+      accessor: InstancesTableAccessor.OS,
```
Comment (junojei) [inline frontend/src/components/sc/components/instances-table/helpers/get-columns.helper.ts:29]:
```suggestion
      Header: InstancesTableHeader.OS,
      accessor: InstancesTableAccessor.OS,
      minWidth: 100,
      width: 200,
```

Comment (junojei) [reply frontend/src/components/sc/components/instances-table/helpers/get-columns.helper.ts:29]: Now we have resizable column.

**Review by junojei — APPROVED** [review]

**Review by what1s1ove — APPROVED** [review]

**Review by Yaroslaww-1 — APPROVED** [review]

**Pull request merged**